libraries/AP_RCProtocol/AP_RCProtocol_SRXL2.cpp

```diff
@@ -0,0 +1,310 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+/*
+  SRXL2 protocol decoder using Horizon Hobby's open source library https://github.com/SpektrumRC/SRXL2
+  Code by Andy Piper
+ */
+
+#include "spm_srxl.h"
+
+#include "AP_RCProtocol.h"
+#include "AP_RCProtocol_SRXL.h"
+#include "AP_RCProtocol_SRXL2.h"
+#include <AP_Math/AP_Math.h>
+#include <AP_Spektrum_Telem/AP_Spektrum_Telem.h>
+#include <AP_Vehicle/AP_Vehicle_Type.h>
+
+extern const AP_HAL::HAL& hal;
+//#define SRXL2_DEBUG
+#ifdef SRXL2_DEBUG
+# define debug(fmt, args...)	hal.console->printf("SRXL2:" fmt "\n", ##args)
+#else
+# define debug(fmt, args...)	do {} while(0)
+#endif
+
+AP_RCProtocol_SRXL2* AP_RCProtocol_SRXL2::_singleton;
+
+AP_RCProtocol_SRXL2::AP_RCProtocol_SRXL2(AP_RCProtocol &_frontend) : AP_RCProtocol_Backend(_frontend)
+{
+    uint32_t uniqueID = AP_HAL::micros();
+
+    _max_channels = MIN((unsigned)SRXL_MAX_CHANNELS,(unsigned)MAX_RCIN_CHANNELS);
```

Comment:
Should be able to determine this at compile-time.

Reply:
Ok